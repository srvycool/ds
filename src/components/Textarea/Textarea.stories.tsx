import React, { useState } from 'react';
import { Textarea } from './Textarea';

export default {
  title: 'Textarea',
  component: Textarea,
};

export const Default: React.FC = () => {
  const [value, setValue] = useState(
    `Aute duis enim culpa nostrud veniam elit occaecat incididunt veniam elit dolor non velit amet. Voluptate nostrud qui cupidatat laborum nisi qui aute. Eu adipisicing mollit proident ad aliquip veniam in. Voluptate occaecat officia sunt quis quis non nostrud sint enim nisi.`
  );

  return (
    <Textarea
      value={value}
      onChange={(event) => setValue(event.target.value)}
    />
  );
};
