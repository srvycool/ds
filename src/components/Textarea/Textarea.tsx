import React from 'react';
import { Box } from '../Box/Box';
import styled, { StyledComponentProps } from 'styled-components';
import { MarginBottomProps } from 'styled-system';
import { Theme } from '../../theme';

export type TextareaProps = StyledComponentProps<'textarea', Theme, {}, never> &
  MarginBottomProps<Theme>;

const StyledBox = styled(Box)`
  transition: 100ms border-color ease-in-out;

  &:focus-within {
    border-color: ${(props) => props.theme.colors.primary500};
  }
`;

const StyledTextarea = styled.textarea`
  ${(props) => props.theme.typography.body1};
  color: ${(props) => props.theme.colors.grey900};
  border: 0;
  width: 100%;
  min-height: 120px;
  padding: 0;

  &:focus {
    outline: none;
  }
`;

export const Textarea: React.FC<TextareaProps> = ({
  marginBottom,
  ...props
}) => {
  return (
    <StyledBox
      boxShadow="primary"
      border="primary"
      padding="s"
      width="100%"
      borderRadius="primary"
      position="relative"
      marginBottom={marginBottom}
    >
      <StyledTextarea {...props} />
    </StyledBox>
  );
};
