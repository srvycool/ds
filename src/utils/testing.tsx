import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
