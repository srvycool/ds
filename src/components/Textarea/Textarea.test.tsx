import React from 'react';
import { render } from '../../utils/testing';
import { Textarea } from './Textarea';

describe('Textarea', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Textarea
        value="Non culpa ullamco est laboris pariatur."
        onChange={() => null}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
