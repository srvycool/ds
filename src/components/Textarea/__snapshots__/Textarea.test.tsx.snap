// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should match snapshot 1`] = `
.c0 {
  box-shadow: 0px 1px 2px rgba(0,0,0,0.06),0px 1px 3px rgba(0,0,0,0.1);
  border: 2px solid #E6DFE1;
  padding: 12px;
  width: 100%;
  border-radius: 8px;
  position: relative;
  box-sizing: border-box;
  -webkit-transition: 100ms border-color ease-in-out;
  transition: 100ms border-color ease-in-out;
}

.c0:focus-within {
  border-color: #DA4167;
}

.c1 {
  font-size: 16px;
  line-height: 1.2;
  font-family: Montserrat;
  color: #493C3F;
  border: 0;
  width: 100%;
  min-height: 120px;
}

.c1:focus {
  outline: none;
}

<div>
  <div
    class="sc-AxjAm c0"
    width="100%"
  >
    <textarea
      class="c1"
    >
      Non culpa ullamco est laboris pariatur.
    </textarea>
  </div>
</div>
`;
